overlord/state/notices.go

```diff
@@ -82,6 +82,9 @@ type Notice struct {
 	// The repeatAfter duration must be less than this, because the notice
 	// won't be tracked after it expires.
 	expireAfter time.Duration
+
+	// Internal data kept for stateful tracking throughout the notice's lifetime.
+	internalData customData
```

Comment:
my plan was literally to add here:

repeatCheckData interface

not full custom data support?

unless your thinking is to be prepared if we want to be ready if we in the end decide to add full internal data here?

Reply:
I was aiming for the more generic internalData to be prepared if this statful callbacks concept is needed in other situations later since this touches state and cannot be easily changed later.

OTOH, I agree that being that generic might not be that beneficial since notices are very limited, and we can add more state if need later.

Reply:
sorry, so I think the simplest thing would be to have here:

repeatCheckData *json.RawMessage

and a GetRepeatCheckValue(value interface) method on Notice ?

and pass to RepeatCheck(oldNotice *Notice, newNoticeOpts *AddNoticeOptions) ?

Reply:
This is very elegant, I like it, Thanks!

Reply:
Updated, Thanks for the feedback!

Reply:
Since RepeatCheck could practically update all options now, I added checks in test that other option fields are updated properly.